### PR TITLE
Features/handle features submit and verified

### PR DESCRIPTION
**Pull request by SamirHassoun**
1) Added Assess Custom HttpStatusCode (similar to Verify)
2) Added popup warning for Assess (similar to Verify)
3) Assess save\Done\ConfirmedDone added in a switch-case format (similar to Verify)
4) PageValidation now checks for both validation errors and warnings
5) PageValidation added checks for DataImpact's Features Submitted and Verified
6) code tidy

### CODE REVIEW

File: src/Portal/Portal/Helpers/PageValidationHelper.cs [hunk 1]
```diff
@@ -165,31 +168,48 @@ public PageValidationHelper(WorkflowDbContext dbContext, HpdDbContext hpdDbConte
...
+        /// <param name="dataImpacts"></param>
+        /// <param name="validationWarningMessages"></param>
+        /// <returns></returns>
+        public bool CheckAssessPageForWarnings(
+            string action,
+            List<DataImpact> dataImpacts,
+            List<string> validationWarningMessages)
+        {
+
+           return !CheckDataImpactFeatures(action, WorkflowStage.Assess, dataImpacts, validationWarningMessages);
```
Comment (Rossall82) [inline src/Portal/Portal/Helpers/PageValidationHelper.cs]: Would it be better to just return the bool result of CheckDataImpactFeatures, and not use the !, as I think the method returns a true or false depending on if there are warnings. It'd make it easier to read for me!

File: src/Portal/Portal/wwwroot/js/Verify.js [hunk 2]
```diff
@@ -60,76 +61,92 @@
...
+                $.ajax({
+                    type: "POST",
+                    url: "Verify/?handler=RejectVerify",
+                    beforeSend: function(xhr) {
+                        xhr.setRequestHeader("RequestVerificationToken",
+                            $('input:hidden[name="__RequestVerificationToken"]').val());
+                    },
+                    data: { "processId": processId, "comment": comment },
+                    complete: function() {
+                        //Add a delay to account for the modalWaitReviewDone modal
```
Comment (Rossall82) [inline src/Portal/Portal/wwwroot/js/Verify.js]: Can we replace modalWaitReviewDone with the correct modal name, just to avoid confusion.

Comment (SamirHassoun) [reply src/Portal/Portal/wwwroot/js/Verify.js]: that comment was there for a long time Ross

Comment (SamirHassoun) [reply src/Portal/Portal/wwwroot/js/Verify.js]: Fixed, thanks

File: src/Portal/Portal/Helpers/PageValidationHelper.cs (hunk 1 above)
Comment (SamirHassoun) [reply src/Portal/Portal/Helpers/PageValidationHelper.cs]: that would break consistency. All validation are returned that way

**Pull request merged**